### DISCUSSION TITLE
Variable load

### DISCUSSION THREAD

**Discussion by msalvati1997** [pybamm-team/PyBaMM#2572] [Q&A]
Hello,
I should use PYBAMM to model a generic battery to which a variable load is subjected. 
The variable load can be expressed in Watt or Wh. 
What I should do is to check the battery status every minute by getting SOC (%) and remaining battery life time, considering the variable load.  How is it possible to do this?  

Thank you in advance.

**Comment by brosaplanella** [comment]
You can define a power drive cycle following [this notebook](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/scripts/experiment_drive_cycle.py). Then you can analyse the solution to obtain the SoC and others, but bear in mind that to have capacity fading you will need to include degradation mechanisms into the model.

**Reply by msalvati1997** [reply]
What is meant by degradation mechanisms? Thank you in advance.

**Reply by brosaplanella** [reply]
If you just use a standard battery model (e.g. SPM, DFN...) capacity won't fade because there is nothing to cause it. These additional effects are called degradation mechanisms. You can read more about that in [this review article](https://iopscience.iop.org/article/10.1088/2516-1083/ac7d31).
